src/V3Const.cpp

```diff
@@ -716,7 +718,10 @@ class ConstBitOpTreeVisitor final : public VNVisitorConst {
                 }
                 // Set width and widthMin precisely
                 resultp->dtypeChgWidth(resultWidth, 1);
-                for (AstNode* const termp : termps) termp->deleteTree();
+                for (AstNode* const termp : termps) {
+                    V3Const::s_containsMemberAccess.erase(termp);
+                    termp->deleteTree();
```

Comment:
How do you avoid having to call erase() at every other deleteTree?
It seems like forgetting to erase the map in the right place is a likely source of bugs.
(Or the pushDelete as described previously (as separate pre-pull, and when it executes this map is cleared).

Alternatively pull in master, and you'll see I had to redo VIsCached to allow clearing, which fixes the problem that got us to needing the std::map here, so perhaps that can now be used closer to your original idea's pull.

Reply:
It is a static method. I can't use `pushDeletep` there.

Reply:
There are only two other calls of `deleteTree` in that file. They are in static methods too. I think we can't replace them.

Reply:
1.  static bool matchOrAndNot(AstNodeBiop* nodep)

can trivially be renamed non-static, and so use pushDelete

2.  static bool matchOrAndNot(AstNodeBiop* nodep)

two places.   Have it take in a reference to the VNDeleter object from the caller.  As part of this, having VNDeleter as an is-a of the Visitor class seems wrong, change that to a has-a, then it's easier to pass around.  It might be that VNDeleter isn't needed in VNVisitorConst but instead only in VNVisitor depending what uses it.

If you agree please make a pre-pull.

Reply:
I agree and created a PR: https://github.com/verilator/verilator/pull/4573.
If everything is ok, I will add necessary changes to this PR when that one is merged.

Reply:
Done.